syntax = "proto3";

package limestone.grpc.proto;

// Represents a single WAL history record.
message BranchEpoch {
  uint64 epoch = 1;        // Epoch value
  uint64 identity = 2;     // identity of the epoch
  int64 timestamp = 3;     // UNIX timestamp, seconds since 1970-01-01T00:00:00Z
}
    
// Request message for listing WAL history records.

message WalHistoryRequest {
  uint32 version = 1;      // Message version for future compatibility. Use value 1 for current implementation.
}

// Response message containing a list of WAL history records.

message WalHistoryResponse {
  uint64 last_epoch = 1;   // the last epoch number
  repeated BranchEpoch records = 2;
}

// Service for retrieving WAL history records.
service WalHistoryService {
  // Returns the WAL history records.
  rpc GetWalHistory(WalHistoryRequest) returns (WalHistoryResponse);
}
